src/qp/interface/qptransform.c

```diff
@@ -283,7 +283,7 @@ PetscErrorCode QPTEnforceEqByProjector(QP qp)
     TRY( MatCreateProd(comm,3,A_arr,&newA) );
   }
   TRY( QPSetOperator(child,newA) );
-  TRY( QPSetOperatorNullSpace(child,qp->R) );
+  //TODO what about null space
```

Comment:
@jkruzik I think here it is easy. `P = I - B'*inv(B*B')*B` so the columns of `B'` span nullspace of `P`, right? In "space language" Im P = Ker B which is orthogonal to Ker P = Im B'.

Reply:
Agreed, but what is Ker P'KP?

Reply:
Since P is a projector and K is multiplied from both sides by P, I believe Ker PKP = Ker P, and the latter is derived above.

Reply:
I think Ker PKP is a union of Ker P, Ker KP, and Ker PKP. For instance, Ker KP = Ker K intersect Image P. I.e., Ker PKP must depend on Ker K. Indeed if K is a matrix of all zeros, Ker PKP = R^n = Ker K.

Reply:
Yes, I agree Ker PKP must depend on Ker K. My [previous reply](https://github.com/permon/permon/pull/47#discussion_r433734399) is nonsensical. `MatCheckNullSpaceMat` revealed that as well. I think I will just leave a TODO again for now.